lib/OpenCloud/LoadBalancer/Service.php

```diff
@@ -51,11 +52,20 @@ public function loadBalancer($id = null)
      */
     public function loadBalancerList($detail = true, array $filter = array())
     {
+        $options = $this->makeResourceIteratorOptions($this->resolveResourceClass('LoadBalancer'));
+
+        if (isset($filter['limit'])) {
+            $options['limit.page'] = $filter['limit'];
+            unset($filter['limit']);
+        }
+
         $url = $this->getUrl();
         $url->addPath(Resource\LoadBalancer::resourceName());
         $url->setQuery($filter);
 
-        return $this->resourceList('LoadBalancer', $url);
+        $options += array('baseUrl' => $url, 'key.marker' => 'id');
```

Comment:
Are you sure you want to use `+` here instead of `array_merge`? Asking because they treat key collisions differently. With `+`, elements from `$options` will be used in case of key collisions whereas with `array_merge`, elements from `array('baseUrl' => ...)` will be used in case of key collisions.